EIPS/eip-6787.md

```diff
@@ -0,0 +1,91 @@
+
+1. *token*: the address of the token to be withdrawn
+2. *amount*: the amount of the token to be withdrawn
+
+The function returns a unique *requestID* that can be used to submit the forced withdrawal using the commitForceWithdraw function.
+
+#### `commitForceWithdraw`
+
+`function commitForceWithdraw(uint256 requestID) external;`
+
+The **commitForceWithdraw** function is used to execute a forced withdrawal operation after the conditions are met. The function takes a *requestID* parameter, which specifies the ID of the forced withdrawal request to be executed. The request must have been previously initiated using the prepareForceWithdraw function.
+
+### Events
+
+#### `PrepareForceWithdraw`
+
+MUST trigger when user successful call to PrepareForceWithdraw.
+
+`event PrepareForceWithdraw(address indexed user, address indexed tokenAddress, uint256 amount);`
+
+## Rationale
+
+The two-phase commit protocol is an important distributed consistency protocol, aiming to ensure data security and consistency in distributed systems. In the Layer2 order book DEX system, to enhance user experience and ensure financial security, we adopt a 1:1 reserve strategy, combined with a decentralized clearing and settlement interface, and a forced withdrawal function to fully guarantee users' funds.
+
+However, such design also faces potential risks. When users engage in perpetual contract transactions, they may incur losses. In this situation, malicious users might exploit the forced withdrawal function to evade losses. To prevent this kind of attack, we propose a two-phase forced withdrawal mechanism.
+
+By introducing the two-phase forced withdrawal function, we can protect users' financial security while ensuring the security of the exchange's assets. In the first phase, the system will conduct a preliminary review of the user's withdrawal request to confirm the user's account status. In the second phase, after the forced withdrawal inspection period, users can directly submit the forced withdrawal request to complete the forced withdrawal process. In this way, we can not only prevent users from exploiting the forced withdrawal function to evade losses but also ensure the asset security for both the exchange and the users.
+
+In conclusion, by adopting the two-phase commit protocol and the two-phase forced withdrawal function, we can effectively guard against malicious behaviors and ensure data consistency and security in distributed systems while ensuring user experience and financial security.
```

Comment:
So this looks like it's justifying why we need two-phase withdrawals, which is the point of the whole EIP. This kind of information belongs in the motivation section.

The rationale should describe individual technical choices made within the EIP itself.

Reply:
Yes, an account may initiate multiple forced withdrawal requests in parallel, so requestid is required.

Reply:
Yes, an account may initiate multiple forced withdrawal requests in parallel, so requestid is required.

Reply:
Perfect. If you move the current rationale into the motivation section and add that note about `requestId` in the rationale, that should be it before merging as draft.

Reply:
Thank you for your reply, I have fix it.